text/1008-tls-and-alpn.rst

```diff
@@ -0,0 +1,580 @@
+        "tls_certdata": "-----BEGIN CERTIFICATE-----\nMIICvjCCAaagAw..."
+    }
+
+And then the client logic for server certificate verification is just
+the same as for local development as explained earlier in this section.
+
+    Open Question: The server may be exposing a chain of certificates.
+    We probably want to balance between convenience (trusting root or
+    intermediate certificate) and safety (trusting only the leaf
+    certificate). Do we want to let the user choose which certificate to
+    trust?
+
+
+ALPN and Protocol Changes
+=========================
+
+The ALPN support in target programming languages:
+
+* Python [4]_: ``set_alpn_protocols()`` and ``selected_alpn_protocol()``
+* Go [5]_: ``SupportedProtos`` and ``NegotiatedProtocol``
+* Node.js [6]_: ``ALPNProtocols`` and ``alpnProtocol``
+
+For now, the EdgeDB server will advertise two protocols in ALPN (however
+EdgeDB is not limited to only these two for future possibilities):
+
+* ``edgedb-binary``: The EdgeDB binary protocol
+* ``http/1.1``: HTTP-based protocol, including the server system API,
+  and extensions like EdgeQL over HTTP, GraphQL over HTTP and Notebook.
+
+The client (including the language bindings and the REPL) should choose
+between ``edgedb-binary`` and ``http/1.1`` during TLS handshake based on
+the scenario in which the user is using the client. If the client didn't
+join the protocol negotiation (e.g. using curl to access the server
+stats endpoint), the server will fallback to ``http/1.1`` - then it is
+literally just HTTPS.
+
+    Note: the server cannot tell if the client asked for a protocol that
+    is not supported by the server, or didn't join the ALPN at all. The
+    server will use ``http/1.1`` for both cases. However if the client
+    asked for a specific protocol, it must check the ALPN result and
+    raise an error if the result is not the expected protocol.
+
+The EdgeDB server will no longer check the magical first-byte to switch
+between HTTP protocol and the binary protocol - it is fully replaced by
+the ALPN negotiation. Once the protocol is agreed upon, there is
+currently no way to switch to another protocol except for reconnecting.
+
+
+Advanced TLS Settings
+=====================
+
+Usually TLS just work out of the box with the default settings. But for
+special security reasons, optionally the advanced TLS settings can be
+modified in the EdgeDB config system per instance. Specifically:
+
++-------------------------+--------------------------+--------------------------------------------------------------------------+
+| EdgeDB Config           | Python SSLContext member | Possible Values                                                          |
++=========================+==========================+==========================================================================+
+| ``tls_minimum_version`` | ``minimum_version``      | ``1.0``, ``1.1``, ``1.2``, ``1.3``, ``MIN_SUPPORTED``, ``MAX_SUPPORTED`` |
```

Comment:
TLS <1.2 isn't considered to be secure anymore, so I think we should have 1.2 as the minimum.